test/app-luatest/gh_9806_http_client_version_test.lua

```diff
@@ -0,0 +1,95 @@
+local t = require('luatest')
+local treegen = require('test.treegen')
+local justrun = require('test.justrun')
+local socket = require('socket')
+
+local g = t.group()
+
+g.before_all(treegen.init)
+g.after_all(treegen.clean)
+
+g.before_each(function(g)
+    g.server = socket.tcp_server('127.0.0.1', 0, function(_s) end)
+end)
+
+g.after_each(function(g)
+    g.server:close()
+end)
+
+local http_request_script = string.dump(function()
+    local http_client = require('http.client').new()
+    local proto = os.getenv('PROTO')
+    local port = os.getenv('HTTP_SERVER_PORT')
+    local uri = proto .. '://127.0.0.1:' .. port
+    pcall(http_client.get, http_client, uri, {
+        verbose = true,
+        http_version = os.getenv('HTTP_VERSION'),
+    })
+end)
+
+g.test_http_version = function()
+    local dir = treegen.prepare_directory(g, {}, {})
+    local script_name = 'http_request.lua'
+    treegen.write_script(dir, script_name, http_request_script)
+
+    local args = {script_name}
+    local opts = {nojson = true, stderr = true}
+    local env = {HTTP_SERVER_PORT = g.server:name().port}
+    env.PROTO = 'http'
```

Comment:
What is a point to parameterize PROTO if it is always constant?

Reply:
It is `http` or `https`.